### PR TITLE
CI: Update base ubuntu to jammy (2022 LTS)

### PR DESCRIPTION
**Pull request by SkySkimmer**

### CODE REVIEW

**Comment by SkySkimmer** [comment]
How are we supposed to get docker-tagged runners these days?  coq-docker-machine-driver  is paused for some reason.
cc @maximedenes @Zimmi48

File: tools/dune_rule_gen/coq_module.ml [hunk 1]
```diff
@@ -53,7 +53,6 @@ let base_obj_files ~rule coq_module =
   let aux_objs = if quick then []
     else
       [ mod_to_obj coq_module ~ext:".glob"
-      ; "." ^ mod_to_obj coq_module ~ext:".aux"
```
Comment (ejgallego) [inline tools/dune_rule_gen/coq_module.ml:56]: That belongs on a separate PR

**Review by ejgallego — COMMENTED** [review]
Why? 22.04 is pretty new, that means dropping testing for many older distros.

File: tools/dune_rule_gen/coq_module.ml (hunk 1 above)
Comment (SkySkimmer) [reply tools/dune_rule_gen/coq_module.ml:56]: oops

**Comment by Zimmi48** [comment]
> How are we supposed to get docker-tagged runners these days?  coq-docker-machine-driver  is paused for some reason.
> cc @maximedenes @Zimmi48 

Reenable it and we'll see if it works or fails.

**Comment by SkySkimmer** [comment]
>Why? 22.04 is pretty new, that means dropping testing for many older distros.

cf https://github.com/mit-plv/bedrock2/issues/288#issuecomment-1286147872

------

It failed at https://gitlab.com/SkySkimmer/coq/-/jobs/3205926426 and I have no more time for messing with this for now so closing.

**Comment by ejgallego** [comment]
> cf [mit-plv/bedrock2#288 (comment)](https://github.com/mit-plv/bedrock2/issues/288#issuecomment-1286147872)

Ouch; maybe we can just create a python env for Coq? I understand that allows installation of newer python quite modularly.
